mathics/core/expression.py

```diff
@@ -1382,14 +1382,26 @@ def __getnewargs__(self):
         return (self._head, self._leaves)
 
 
+# This function allows to avoid a circular reference when we want
+# to create `Expression` objects in the `mathics.core.symbols` and `mathics.core.atoms` modules.
+
+
 def _create_expression(self, head, *leaves):
     return Expression(head, *leaves)
 
 
 BaseExpression.create_expression = _create_expression
```

Comment:
The type in BaseExpression is `Any`, but here it should be `Callable` or possibly `Optional[Callable]`. 

As I look over code there is something that doesn't smell right. But if this is the current workaround it _should_ be in a docstring because that is what it's function is: anyone who wants to understand why this exists needs to be informed that it is to avoid a circular reference or is some sort of workaround that might (or should) be removed. That is its purpose.